### ISSUE TITLE
dhclient needed for network manager

### ISSUE THREAD

**Issue by jmcgrath207** [Antergos/Cnchi#807]
### Problem:
During install, dhclient is not included and network manager will not work without it. Once I installed dhclient, the network manager worked correctly.

dhcpcd is installed and worked when I built a wpa_supplicant profile but for some reason, network manager chooses to ignore this dhcp client.

### Log Files

```
# sudo journalctl -u NetworkManager -b
- Logs begin at Sat 2017-10-21 14:33:14 CDT, end at Wed 2017-10-25 23:27:29 CDT. --
Oct 25 23:13:11 john-home systemd[1]: Starting Network Manager...
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2226] NetworkManager (version 1.8.4-1, Arch Linux) is starting... (for the first time)
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2227] Read config: /etc/NetworkManager/NetworkManager.conf (lib: 20-connectivity.conf)
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2320] manager[0x560d4bf7d100]: monitoring kernel firmware directory '/usr/lib/firmware'.
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2337] policy: hostname management mode: default
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2339] dns-mgr[0x560d4bf89960]: init: dns=default, rc-manager=resolvconf
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2349] rfkill1: found WiFi radio killswitch (at /sys/devices/pci0000:00/0000:00:1c.1/0000:03:00.0/ieee80211/phy0/rfkill1) (driver iwlwifi)
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2351] manager[0x560d4bf7d100]: rfkill: WiFi hardware radio set enabled
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2351] manager[0x560d4bf7d100]: rfkill: WWAN hardware radio set enabled
Oct 25 23:13:11 john-home systemd[1]: Started Network Manager.
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2856] settings: loaded plugin keyfile: (c) 2007 - 2016 Red Hat, Inc.  To report bugs please use the NetworkManager mailing list.
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2866] settings: loaded plugin iBFT: (c) 2014 Red Hat, Inc.  To report bugs please use the NetworkManager mailing list. (/usr/lib/NetworkManager/libnm-settings-plugin-ibft.so)
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.2982] keyfile: new connection /etc/NetworkManager/system-connections/Auto flightpath-2G (69b2fdfd-a460-4fb9-9ab3-865d21e3642b,"Auto flightpath-2G")
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.3096] keyfile: new connection /etc/NetworkManager/system-connections/Auto flightpath-5G (6659e2aa-ac31-49d4-bc43-8b302dd1e5cd,"Auto flightpath-5G")
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.3226] keyfile: new connection /etc/NetworkManager/system-connections/Tomato24 (e50b8646-249b-444e-bd57-789cafa4120c,"Tomato24")
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.4692] settings: hostname: using hostnamed
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.4692] settings: hostname changed from (none) to "john-home"
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.4706] dhcp-init: default DHCP client 'dhclient' is not available
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.4707] dhcp-init: Using DHCP client 'internal'
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.4707] manager: rfkill: WiFi enabled by radio killswitch; enabled by state file
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.4707] manager: rfkill: WWAN enabled by radio killswitch; enabled by state file
Oct 25 23:13:11 john-home NetworkManager[543]: <info>  [1508991191.4708] manager: Networking is enabled by state file
```

**Comment by killajoe** [comment]
"During install, dhclient is not included and network manager will not work without it."
Not clear or not true... i do not have unworking networkmanager during install. 
And GNOME install have the dhclient after installation installed.

**Comment by lots0logs** [comment]
NetworkManager uses its own internal client. That is the intended behavior.

**Comment by jmcgrath207** [comment]
I forgot to clarify I installed Cinnamon, not GNOME. I could not get NetworkManager Working with without running installing dh-client from the arch repo.

**Comment by lots0logs** [comment]
Perhaps there is something causing the issue that is local to your network or gateway. In the majority of cases NetworkManager doesn't need to use dhclient. Though in cases where there is some issue, it can fallback to dhclient if its installed.

**Comment by jmcgrath207** [comment]
To make this work I would kill the network manager and the  WPA supplicant process. Then I would re-run the WPA supplicant against my custom profile config and manually run the dhcpd process to grab an Ip address and DNS information. This would result in the wireless connection working correctly. 

I also ran a test where I would connect with the network manager and ran dhcpd process in the shell and it would work for a few minutes before being locally Deauth.

However, once I installed the dhclient on my box, I was able to connect with my network manager successful without issue. 

The issue to me seems to be network manager not properly using the dhcpd process when it should.
